### ISSUE TITLE
Mutation is a debuggers worst nightmare

### ISSUE THREAD

**Issue by Akin909** [fac-13/KAEA#17]
[Here](https://github.com/fac-13/KAEA/blob/adb9fa2cf1c4718cc711ffae7e567ac870ceb367/src/components/quiz.js#L17)
There is a sacred law in the land of react think of it as the first commandment *Never* mutate state 😉 or props you should always be v. wary of code that looks like
```js
	thing1 = "apple"
	thing1 = randomOtherThing

// More scary is
	this.state.apple = "pear"
```

The reason mutation in this (and many other scenarios) is a big problem is that by reassigning state or mutating state you immediately lose the ability to track what things were. think of it a little like git history if you go back to a previous commit and say delete it or change what happened then your git history becomes hard to follow since there are missing bits, it becomes unclear what happened or how you even got there

**Comment by oliverjam** [comment]
I was confused by this function as well. You don't actually have an `oldScore` key in your state, so the destructuring isn't doing anything. It would be a bit clearer if you made what you're doing explicity:

```js
  score = () => {
    console.log('we have scored');
    const { score: oldScore } = this.state;
    this.setState({ score: oldScore + 10 });
  };
```
